lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb

```diff
@@ -1,221 +0,0 @@
-      end
-    end
-
-    # Drop table and optionally disable triggers.
-    # Changes adapted from https://github.com/matthuhiggins/foreigner/blob/e72ab9c454c156056d3f037d55e3359cd972af32/lib/foreigner/connection_adapters/sql2003.rb
-    # NOTE: Disabling referential integrity requires superuser access in postgres.
-    #       Default AR behavior is just to drop_table.
-    #
-    # == Options:
-    # * :force - force disabling of referential integrity
-    #
-    # Note: I don't know a good way to test this -mike 20120420
-    def drop_table(*args)
-      options = args.clone.extract_options!
-      if options[:force]
-        disable_referential_integrity { super }
-      else
-        super
-      end
-    end
-
-    # Add foreign key.
-    #
-    # Ensures that an index is created for the foreign key, unless :exclude_index is true.
-    #
-    # == Options:
-    # * :column
-    # * :primary_key
-    # * :dependent
-    # * :exclude_index    [Boolean]
-    # * :concurrent_index [Boolean]
-    #
-    # @param [String, Symbol]          from_table
-    # @param [String, Symbol]          to_table
-    # @param [Hash]                    options
-    # @option options [String, Symbol] :column
-    # @option options [String, Symbol] :primary_key
-    # @option options [Hash]           :dependent
-    # @option options [Boolean]        :exclude_index
-    # @option options [Boolean]        :concurrent_index
-    #
-    # @raise [PgSaurus::IndexExistsError] when :exclude_index is true, but the index already exists
-    def add_foreign_key(from_table, to_table, options = {})
-      options[:column]             ||= id_column_name_from_table_name(to_table)
-      options[:exclude_index]      ||= false
-
-      if index_exists?(from_table, options[:column]) && !options[:exclude_index]
-        raise PgSaurus::IndexExistsError,
-          "The index, #{index_name(from_table, options[:column])}, already exists." \
-          "  Use :exclude_index => true when adding the foreign key."
-      end
-
-      sql = "ALTER TABLE #{quote_table_name(from_table)} #{add_foreign_key_sql(from_table, to_table, options)}"
-      execute(sql)
-
-      # GOTCHA:
-      #   Index can not be created concurrently inside transaction in PostgreSQL.
-      #   So, in case of concurrently created index with foreign key only
-      #   foreign key will be created inside migration transaction and after
-      #   closing transaction queries for index creation will be send to database.
-      #   That's why I prevent here normal index creation in case of
-      #   `concurrent_index` option is given.
-      #   NOTE: Index creation after closing migration transaction could lead
-      #   to weird effects when transaction moves smoothly, but index
-      #   creation with error. In that case transaction will not be rolled back.
-      #   As it was closed before even index was attempted to create.
-      #   -- zekefast 2012-09-12
```

Comment:
No. What we were doing before was silently not creating the index. Not what I want to do going forward.